spec/features/visitor_signs_up_spec.rb

```diff
@@ -0,0 +1,30 @@
+require 'spec_helper'
+require 'rails_helper'
+
+feature 'Visitor signs up' do
+  scenario 'with valid email and password' do
+    sign_up_with 'valid@example.com', 'password'
+
+    expect(page).to have_content('Logout')
+  end
+
+  scenario 'with invalid email' do
+    sign_up_with 'invalid_email', 'password'
+
+    expect(page).to have_content('Sign up')
+  end
+
+  scenario 'with blank password' do
+    sign_up_with 'valid@example.com', ''
+
+    expect(page).to have_content('Sign up')
+  end
```

Comment:
In all three tests above I recommend not to expect to have a link/button with name (because this name dosn't depend user action). It's better to expect e.x. a proper error message.